tests/Get-DbaSqlInstanceProperty.Tests.ps1

```diff
@@ -0,0 +1,44 @@
+$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
+. "$PSScriptRoot\constants.ps1"
+
+Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
+    Context "Validate parameters" {
+        $paramCount = 3
+        $defaultParamCount = 11
+        [object[]]$params = (Get-ChildItem function:\Get-DbaSqlInstanceProperty).Parameters.Keys
+        $knownParameters = 'Computer', 'SqlInstance', 'SqlCredential', 'Credential', 'EnableException'
+        It "Should contain our specific parameters" {
+            ( (Compare-Object -ReferenceObject $knownParameters -DifferenceObject $params -IncludeEqual | Where-Object SideIndicator -eq "==").Count ) | Should Be $paramCount
+        }
+        It "Should only contain $paramCount parameters" {
+            $params.Count - $defaultParamCount | Should Be $paramCount
+        }
+    }
+}
+
+Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
+    Context "Command actually works" {
+        $results = Get-DbaSqlInstanceProperty -SqlInstance $script:instance2
+        It "Should have correct properties" {
+            $ExpectedProps = 'ComputerName,InstanceName,PropertyType,SqlInstance'.Split(',')
+            (($results | Get-Member -MemberType NoteProperty).name | Sort-Object) | Should Be ($ExpectedProps | Sort-Object)
+        }
+        It "Should return that instance2 is running developer edition" {
+            ($results | Where-Object {$_.name -eq 'Edition'}).Value | Should Be "Developer Edition (64-bit)"
```

Comment:
It would be ideal to make more generic test...values that would/could be the same for both local environments and AppVeyory.

Reply:
Thanks for the feedback, would you suggest picking a different property, maybe IsSingleUser which would more than likely be False on all systems, or  picking ResourceVersionString which returns the build number and running that through `Get-DbaSqlBuildReference` to check it's valid?